bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/TargetGroupConfig.java

```diff
@@ -21,27 +22,29 @@
     public static final String TYPE_VALUE="targetgroup";
 
     public static TargetGroupConfig createBaseGroup(final TargetEnvironment env) {
-        final ImList<SPTarget> targets = env.getUserTargets().map(u -> u.target).cons(env.getArbitraryTargetFromAsterism());
-        return new TargetGroupConfig(TccNames.BASE, targets, ImOption.apply(env.getArbitraryTargetFromAsterism()));
+        final ImList<SPTarget> scienceTargets = env.getAsterism().allSpTargetsJava();
+        final ImList<SPTarget> userTargets    = env.getUserTargets().map(u -> u.target);
+        final ImList<SPTarget> targets        = scienceTargets.append(userTargets);
+        return new TargetGroupConfig(TccNames.BASE, targets, ImOption.apply(env.getPrimaryTargetFromAsterism()));
     }
 
     public static TargetGroupConfig createGuideGroup(final GuideProbeTargets gt) {
         final String tag = TargetConfig.getTag(gt.getGuider());
         final ImList<SPTarget> targets = gt.getTargets();
-        final Option<SPTarget> primaryOpt = gt.getPrimary();
+        final Option<SPSkyObject> primaryOpt = gt.getPrimary().map(t -> (SPSkyObject) t);
         return new TargetGroupConfig(tag, targets, primaryOpt);
     }
 
-    private TargetGroupConfig(final String name, final ImList<SPTarget> targets, final Option<SPTarget> primaryTarget) {
+    private TargetGroupConfig(final String name, final ImList<SPTarget> targets, final Option<SPSkyObject> primaryTarget) {
         super(name);
 
         addAttribute(TYPE, TYPE_VALUE);
 
-        primaryTarget.map(t -> t.getName())
-                .filter(n -> !"".equals(n))
+        primaryTarget.map(SPSkyObject::getName)
+                .filter(n -> !n.isEmpty())
```

Comment:
The previous version would handle a `null` name.  Is that unpossible now?

Reply:
Hmmm. Good point. I'm not certain. I'll check, and change it back if I'm not sure.

Reply:
I don't think it will ever be `null`, but let's not play with 🔥 .